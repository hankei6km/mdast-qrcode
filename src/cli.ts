import { Readable, Writable } from 'stream';
import { fromMarkdown } from 'mdast-util-from-markdown';
import { toMarkdown } from 'mdast-util-to-markdown';
import { toImageDataURL } from './qrcode.js';
import { frontmatter } from 'micromark-extension-frontmatter';
import {
  frontmatterFromMarkdown,
  frontmatterToMarkdown
} from 'mdast-util-frontmatter';

type ToMarkdownOptions = {
  bullet?: '-' | '*' | '+';
  closeAtx?: boolean;
  emphasis?: '_' | '*';
  fence?: '~' | '`';
  fences?: boolean;
  incrementListMarker?: boolean;
  listItemIndent?: 'tab' | 'one' | 'mixed';
  quote?: '"' | "'";
  resourceLink?: boolean;
  rule?: '-' | '_' | '*';
  ruleRepetition?: number;
  ruleSpaces?: boolean;
  setext?: boolean;
  strong?: '_' | '*';
  tightDefinitions?: boolean;
};

type Opts = {
  stdin: Readable;
  stdout: Writable;
  stderr: Writable;
  bullet?: ToMarkdownOptions['bullet'];
  rule?: ToMarkdownOptions['rule'];
};

const cli = async ({
  stdin,
  stdout,
  stderr,
  bullet = '-',
  rule = '-'
}: Opts): Promise<number> => {
  try {
    let source = '';
    await new Promise((resolve) => {
      stdin.on('data', (d) => (source = source + d));
      stdin.on('end', () => resolve(source));
    });
    const tree = fromMarkdown(source, {
      extensions: [frontmatter(['yaml', 'toml'])],
      mdastExtensions: [frontmatterFromMarkdown(['yaml', 'toml'])]
    });
    await toImageDataURL(tree);
    stdout.write(
      toMarkdown(tree, {
        bullet,
        rule,
        extensions: [frontmatterToMarkdown(['yaml', 'toml'])]
      })
    );
  } catch (err: any) {
    stderr.write(err.toString());
    stderr.write('\n');
    return 1;
  }
  return 0;
};

export default cli;
