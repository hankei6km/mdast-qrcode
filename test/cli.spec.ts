import { createReadStream } from 'fs';
import { PassThrough } from 'stream';
import cli from '../src/cli.js';

describe('cli()', () => {
  it('should convert example file', async () => {
    const f = createReadStream('example/qrcode-deck.md');
    const stdin = new PassThrough();
    const stdout = new PassThrough();
    const stderr = new PassThrough();
    let outData = '';
    stdout.on('data', (d) => (outData = outData + d));
    let errData = '';
    stderr.on('data', (d) => (errData = errData + d));
    process.nextTick(() => {
      f.pipe(stdin);
      //stdin.end();
    });
    expect(
      await cli({
        stdin,
        stdout,
        stderr
      })
    ).toEqual(0);
    expect(outData).toMatchSnapshot();
    expect(errData).toEqual('');
  });
  it('should accept options', async () => {
    const stdin = new PassThrough();
    const stdout = new PassThrough();
    const stderr = new PassThrough();
    let outData = '';
    stdout.on('data', (d) => (outData = outData + d));
    let errData = '';
    stderr.on('data', (d) => (errData = errData + d));
    process.nextTick(() => {
      stdin.write(
        '---\ntitle: title2\n---\n\n# test2\n\n- item1\n- item2\n\n---\n\n# page2\n'
      );
      stdin.end();
    });
    expect(
      await cli({
        stdin,
        stdout,
        stderr,
        bullet: '+',
        rule: '*'
      })
    ).toEqual(0);
    expect(outData).toEqual(
      '---\ntitle: title2\n---\n\n# test2\n\n+   item1\n+   item2\n\n***\n\n# page2\n'
    );
    expect(errData).toEqual('');
  });
});
