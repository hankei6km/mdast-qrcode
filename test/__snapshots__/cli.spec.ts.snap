// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli() should convert example file 1`] = `
"---
title: QR code
description: Slide with embedded QR code
theme: gaia
class: gaia
---

<!--
_class:
  - lead
  - gaia
-->

# **QR** code

## mdast-qrcode exmaple slide

https://github.com/hankei6km/mdast-qrcode

![bg left](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJQAAACUCAYAAAB1PADUAAAAAklEQVR4AewaftIAAASwSURBVO3BQY4kRxIEQdNA/f/Lun30U2AT6dWcIU0Ef6RqyUnVopOqRSdVi06qFp1ULTqpWnRSteikatFJ1aKTqkUnVYtOqhadVC06qVp0UrXok5eA/CY1E5BJzQTkCTUTkEnNBGSTmgnIb1LzxknVopOqRSdViz5ZpmYTkDfU3ACZgLyh5gkgT6jZBGTTSdWik6pFJ1WLPvkyIE+oeULNE0CeUPMGkEnNJiBPqPmmk6pFJ1WLTqoWffKXA3KjZlJzA2QC8gSQSc0EZFLzb3JSteikatFJ1aJP/nJqJiATkEnNBGRSMwGp/99J1aKTqkUnVYs++TI1fxIgT6h5A8gNkEnNE2r+JCdVi06qFp1ULfpkGZDfBGRSMwGZ1ExAboBMaiYgk5obNROQJ4D8yU6qFp1ULTqpWvTJS2r+JECeUHOj5gkgm9T8TU6qFp1ULTqpWvTJS0AmNROQGzUTkCfU3AB5A8gmIJOaGyCTmhsgk5oJyI2aN06qFp1ULTqpWoQ/8gKQGzUTkCfUTEBu1ExA3lDzBJA31ExAnlDzBJBJzRsnVYtOqhadVC36ZJmaJ9TcALlRMwG5UXMDZJOaGyATkEnNBGRScwPkRs2mk6pFJ1WLTqoW4Y8sAvKEmj8JkBs1TwC5UTMBuVEzAZnUPAFkUvPGSdWik6pFJ1WL8EdeADKpuQGySc0E5Ak1E5DfpGYTkBs133RSteikatFJ1aJPvgzIpGYCMqm5ATIBmdTcAJmATGomIP8kIJOaGzUTkAnIjZo3TqoWnVQtOqla9MlLam7UPAFkE5BJzQ2QGzU3QJ4AMqn5m51ULTqpWnRStQh/5IuA3Kh5Asgban4TkN+k5gbIpGbTSdWik6pFJ1WLPnkJyKRmUvMEkCfUTECeAHKj5gbIpOYNNU8AmYBMan7TSdWik6pFJ1WLPnlJzQTkRs0E5EbNE2q+CcikZgIyqdkEZFIzAZmA/KaTqkUnVYtOqhbhj7wAZFIzAZnUPAFkUjMBmdRsArJJzQRkUjMBeULNBGRS800nVYtOqhadVC365CU1N2omIJOaGzXfBGRSM6l5AsgTav5NTqoWnVQtOqlahD+yCMik5gbIE2qeAPKGmhsgT6iZgPwmNd90UrXopGrRSdWiT5apeULNG0CeUDMB+ZOoeQLIE0Bu1LxxUrXopGrRSdWiT14C8pvUPKHmRs0E5Ak1E5AbIE8AmdQ8oWYCMqnZdFK16KRq0UnVok+WqdkE5A0gN2omNW+omYBMaiYgN2q+Ccik5o2TqkUnVYtOqhZ98mVAnlDzhJoJyI2aCcikZgJyo2YC8gaQN9RMQH7TSdWik6pFJ1WLPvnLAXkCyA2QJ4BMam6ATGomIJOaCcgNkEnNDZBNJ1WLTqoWnVQt+uQ/Rs0NkBs1N0Bu1ExAJjVPqLkBMqn5ppOqRSdVi06qFn3yZWq+Sc0NkAnIE2omIDdq3gAyqXkCyKRmAjKp2XRSteikatFJ1aJPlgH5TUCeUDMBeUPNDZA3gExqbtT8k06qFp1ULTqpWoQ/UrXkpGrRSdWik6pFJ1WLTqoWnVQtOqladFK16KRq0UnVopOqRSdVi06qFp1ULTqpWvQ/hB8NU+NRNmkAAAAASUVORK5CYII=)

---

## QR code (url)

source:

\`\`\`markdown
![h:200](qrcode:https://github.com/hankei6km/mdast-qrcode)
https://github.com/hankei6km/mdast-qrcode
\`\`\`

yield:

![h:200](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJQAAACUCAYAAAB1PADUAAAAAklEQVR4AewaftIAAASwSURBVO3BQY4kRxIEQdNA/f/Lun30U2AT6dWcIU0Ef6RqyUnVopOqRSdVi06qFp1ULTqpWnRSteikatFJ1aKTqkUnVYtOqhadVC06qVp0UrXok5eA/CY1E5BJzQTkCTUTkEnNBGSTmgnIb1LzxknVopOqRSdViz5ZpmYTkDfU3ACZgLyh5gkgT6jZBGTTSdWik6pFJ1WLPvkyIE+oeULNE0CeUPMGkEnNJiBPqPmmk6pFJ1WLTqoWffKXA3KjZlJzA2QC8gSQSc0EZFLzb3JSteikatFJ1aJP/nJqJiATkEnNBGRSMwGp/99J1aKTqkUnVYs++TI1fxIgT6h5A8gNkEnNE2r+JCdVi06qFp1ULfpkGZDfBGRSMwGZ1ExAboBMaiYgk5obNROQJ4D8yU6qFp1ULTqpWvTJS2r+JECeUHOj5gkgm9T8TU6qFp1ULTqpWvTJS0AmNROQGzUTkCfU3AB5A8gmIJOaGyCTmhsgk5oJyI2aN06qFp1ULTqpWoQ/8gKQGzUTkCfUTEBu1ExA3lDzBJA31ExAnlDzBJBJzRsnVYtOqhadVC36ZJmaJ9TcALlRMwG5UXMDZJOaGyATkEnNBGRScwPkRs2mk6pFJ1WLTqoW4Y8sAvKEmj8JkBs1TwC5UTMBuVEzAZnUPAFkUvPGSdWik6pFJ1WL8EdeADKpuQGySc0E5Ak1E5DfpGYTkBs133RSteikatFJ1aJPvgzIpGYCMqm5ATIBmdTcAJmATGomIP8kIJOaGzUTkAnIjZo3TqoWnVQtOqla9MlLam7UPAFkE5BJzQ2QGzU3QJ4AMqn5m51ULTqpWnRStQh/5IuA3Kh5Asgban4TkN+k5gbIpGbTSdWik6pFJ1WLPnkJyKRmUvMEkCfUTECeAHKj5gbIpOYNNU8AmYBMan7TSdWik6pFJ1WLPnlJzQTkRs0E5EbNE2q+CcikZgIyqdkEZFIzAZmA/KaTqkUnVYtOqhbhj7wAZFIzAZnUPAFkUjMBmdRsArJJzQRkUjMBeULNBGRS800nVYtOqhadVC365CU1N2omIJOaGzXfBGRSM6l5AsgTav5NTqoWnVQtOqlahD+yCMik5gbIE2qeAPKGmhsgT6iZgPwmNd90UrXopGrRSdWiT5apeULNG0CeUDMB+ZOoeQLIE0Bu1LxxUrXopGrRSdWiT14C8pvUPKHmRs0E5Ak1E5AbIE8AmdQ8oWYCMqnZdFK16KRq0UnVok+WqdkE5A0gN2omNW+omYBMaiYgN2q+Ccik5o2TqkUnVYtOqhZ98mVAnlDzhJoJyI2aCcikZgJyo2YC8gaQN9RMQH7TSdWik6pFJ1WLPvnLAXkCyA2QJ4BMam6ATGomIJOaCcgNkEnNDZBNJ1WLTqoWnVQt+uQ/Rs0NkBs1N0Bu1ExAJjVPqLkBMqn5ppOqRSdVi06qFn3yZWq+Sc0NkAnIE2omIDdq3gAyqXkCyKRmAjKp2XRSteikatFJ1aJPlgH5TUCeUDMBeUPNDZA3gExqbtT8k06qFp1ULTqpWoQ/UrXkpGrRSdWik6pFJ1WLTqoWnVQtOqladFK16KRq0UnVopOqRSdVi06qFp1ULTqpWvQ/hB8NU+NRNmkAAAAASUVORK5CYII=)
https://github.com/hankei6km/mdast-qrcode

---

## QR code (alt)

source:

\`\`\`markdown
![h:200:qrcode:https://github.com/hankei6km/mdast-qrcode](mdast-qrcode.png)
https://github.com/hankei6km/mdast-qrcode
\`\`\`

yield:

![h:200](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJQAAACUCAYAAAB1PADUAAAAAklEQVR4AewaftIAAASwSURBVO3BQY4kRxIEQdNA/f/Lun30U2AT6dWcIU0Ef6RqyUnVopOqRSdVi06qFp1ULTqpWnRSteikatFJ1aKTqkUnVYtOqhadVC06qVp0UrXok5eA/CY1E5BJzQTkCTUTkEnNBGSTmgnIb1LzxknVopOqRSdViz5ZpmYTkDfU3ACZgLyh5gkgT6jZBGTTSdWik6pFJ1WLPvkyIE+oeULNE0CeUPMGkEnNJiBPqPmmk6pFJ1WLTqoWffKXA3KjZlJzA2QC8gSQSc0EZFLzb3JSteikatFJ1aJP/nJqJiATkEnNBGRSMwGp/99J1aKTqkUnVYs++TI1fxIgT6h5A8gNkEnNE2r+JCdVi06qFp1ULfpkGZDfBGRSMwGZ1ExAboBMaiYgk5obNROQJ4D8yU6qFp1ULTqpWvTJS2r+JECeUHOj5gkgm9T8TU6qFp1ULTqpWvTJS0AmNROQGzUTkCfU3AB5A8gmIJOaGyCTmhsgk5oJyI2aN06qFp1ULTqpWoQ/8gKQGzUTkCfUTEBu1ExA3lDzBJA31ExAnlDzBJBJzRsnVYtOqhadVC36ZJmaJ9TcALlRMwG5UXMDZJOaGyATkEnNBGRScwPkRs2mk6pFJ1WLTqoW4Y8sAvKEmj8JkBs1TwC5UTMBuVEzAZnUPAFkUvPGSdWik6pFJ1WL8EdeADKpuQGySc0E5Ak1E5DfpGYTkBs133RSteikatFJ1aJPvgzIpGYCMqm5ATIBmdTcAJmATGomIP8kIJOaGzUTkAnIjZo3TqoWnVQtOqla9MlLam7UPAFkE5BJzQ2QGzU3QJ4AMqn5m51ULTqpWnRStQh/5IuA3Kh5Asgban4TkN+k5gbIpGbTSdWik6pFJ1WLPnkJyKRmUvMEkCfUTECeAHKj5gbIpOYNNU8AmYBMan7TSdWik6pFJ1WLPnlJzQTkRs0E5EbNE2q+CcikZgIyqdkEZFIzAZmA/KaTqkUnVYtOqhbhj7wAZFIzAZnUPAFkUjMBmdRsArJJzQRkUjMBeULNBGRS800nVYtOqhadVC365CU1N2omIJOaGzXfBGRSM6l5AsgTav5NTqoWnVQtOqlahD+yCMik5gbIE2qeAPKGmhsgT6iZgPwmNd90UrXopGrRSdWiT5apeULNG0CeUDMB+ZOoeQLIE0Bu1LxxUrXopGrRSdWiT14C8pvUPKHmRs0E5Ak1E5AbIE8AmdQ8oWYCMqnZdFK16KRq0UnVok+WqdkE5A0gN2omNW+omYBMaiYgN2q+Ccik5o2TqkUnVYtOqhZ98mVAnlDzhJoJyI2aCcikZgJyo2YC8gaQN9RMQH7TSdWik6pFJ1WLPvnLAXkCyA2QJ4BMam6ATGomIJOaCcgNkEnNDZBNJ1WLTqoWnVQt+uQ/Rs0NkBs1N0Bu1ExAJjVPqLkBMqn5ppOqRSdVi06qFn3yZWq+Sc0NkAnIE2omIDdq3gAyqXkCyKRmAjKp2XRSteikatFJ1aJPlgH5TUCeUDMBeUPNDZA3gExqbtT8k06qFp1ULTqpWoQ/UrXkpGrRSdWik6pFJ1WLTqoWnVQtOqladFK16KRq0UnVopOqRSdVi06qFp1ULTqpWvQ/hB8NU+NRNmkAAAAASUVORK5CYII=)
https://github.com/hankei6km/mdast-qrcode

---

## QR code (options)

source:

\`\`\`markdown
![h:200:qrcode:https://github.com/hankei6km/mdast-qrcode](mdast-qrcode-color_dark-563EBFF.png)
https://github.com/hankei6km/mdast-qrcode
\`\`\`

yield:

![h:200](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJQAAACUCAYAAAB1PADUAAAAAklEQVR4AewaftIAAAWGSURBVO3B0ZElSQpFwVNhTwRkQDLEQjJkQIfe/uQrbNKSqumeve5fv35DZMlBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNFBZNGHlzyan1RpTB7NVGlMHs0Tlcbk0UyVxuTRbKo0Jo/mJ1UabxxEFh1EFh1EFn1YVmls8mjeqDRuPJrJo3mj0njCo3mi0tjk0Ww6iCw6iCw6iCz68M08micqjScqjSc8micqjTc8mqnS2OTRPFFpfKeDyKKDyKKDyKIPfzmP5qbSmCqNG49m8mie8GimSmPyaKZK47/kILLoILLoILLow1+u0pg8msmjmSqNyaOZKo3Jo5F/7iCy6CCy6CCy6MM3qzT+JB7NE5XGGx7NjUczVRpPVBp/koPIooPIooPIog/LPJqf5NFMlcbk0UyVxuTR3Hg0U6UxeTRTpXFTaUwezRMezZ/sILLoILLoILLow0uVxp/Eo3mi0ripNJ7waDZVGn+Tg8iig8iig8iiDy95NFOlMXk0N5XG5NE8UWnceDRveDSbPJqp0rjxaKZK48ajmSqNyaO5qTTeOIgsOogsOogs+vr1Gy94NDeVxuTRPFFpTB7NTaUxeTRvVBpPeDRvVBqTR/NEpfGERzNVGm8cRBYdRBYdRBZ9WFZpPFFp3Hg0N5XG5NHcVBo3Hs2mSuPGo5k8mqnSmDyaqdK48WhuKo1NB5FFB5FFB5FFX79+Y5FH80Sl8SfxaG4qjSc8mptKY/JobiqNyaOZKo0nPJqp0njjILLoILLoILLo69dvvODRTJXGjUezqdKYPJonKo3Jo/lJlcYmj+am0vhOB5FFB5FFB5FFH76ZRzNVGpNHM1UaNx7N5NFMlcaNRzN5NFOlMXk0/yaPZqo0biqNyaOZPJqbSuONg8iig8iig8iiDy9VGjeVxhMezSaPZqo0bjyam0rjxqN5wqOZKo2/2UFk0UFk0UFk0dev3/hGHs1NpfGER/NGpfGTPJqfVGnceDRTpbHpILLoILLoILLow0sezVRpTJXGEx7NE5XG5NE84dHcVBo3Hs1UabxRaTzh0UwezVRp/KSDyKKDyKKDyKIPL1Uak0dzU2lMHs1NpfFEpfGdPJqp0pg8mqnS2OTRTJXG5NFMHs1POogsOogsOogs+vr1Gy94NFOlMXk0U6XxhEczVRqTRzNVGps8mk2VxuTRTJXG5NE8UWlMHs1UaXyng8iig8iig8iiDy9VGjeVxuTRTJXGTaXxnTyaqdKYKo0nPJonKo3/koPIooPIooPIog/LPJqp0pgqjcmjeaLSmCqNG4/mCY9mqjRuPJo3PJpNHs2/6SCy6CCy6CCy6MOySuOJSuMNj+aJSmPyaP4klcYTHs0THs1NpfHGQWTRQWTRQWTRh5c8mp9UaTxRadxUGpNH80SlMXk0Nx7NEx7NVGk8UWlMHs1UaWw6iCw6iCw6iCz6sKzS2OTRvOHR3FQaU6XxRqUxeTRTpTF5NDeVxnfyaKZK442DyKKDyKKDyKIP38yjeaLSeKLSmDyam0pj8mimSmPyaG4qjcmjecOjeaPSmDyan3QQWXQQWXQQWfThL+fRPOHR3Hg0T3g0U6Vx49FMlcbk0UyVxuTR3Hg0U6Vx49FsOogsOogsOogs+vB/ptK48WhuKo0bj+am0pg8mqnSeKLSuPFopkrjOx1EFh1EFh1EFn34ZpXGd6o0bjyayaN5otKYPJqbSuMNj2aqNJ7waKZKY/Jopkpj00Fk0UFk0UFk0YdlHs1P8mieqDQmj+aNSuPGo3nDo5kqjZtK4990EFl0EFl0EFn09es3RJYcRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBYdRBb9D6YsDyyqSWNRAAAAAElFTkSuQmCC)
https://github.com/hankei6km/mdast-qrcode

---

## QR code (error creection level = H)

source:

\`\`\`markdown
![h:200:qrcode:https://github.com/hankei6km/mdast-qrcode](mdast-qrcode-color_dark-563EBFF-eclevel-H.png)
https://github.com/hankei6km/mdast-qrcode
\`\`\`

yield:

![h:200](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAALQAAAC0CAYAAAA9zQYyAAAAAklEQVR4AewaftIAAAjRSURBVO3BwbFcu45FwSVGmQAbYBnM2pbBBvigryF6wgjGOXWfxEbmr99/MMYlFmNcZDHGRRZjXGQxxkUWY1xkMcZFFmNcZDHGRRZjXGQxxkUWY1xkMcZFFmNcZDHGRRZjXOTDQx7FT0oZnUfRpYzOo9hJGf8Sj6JLGZ1H0aWMJzyKn5QynliMcZHFGBdZjHGRDy9LGW/yKE54FDspY8ej6FLGjkfxRMrY8Si6lNF5FDsexYmUsZMy3uRRvGkxxkUWY1xkMcZFPnyZR3EiZTyRMjqP4kTK2PEodlJG51F0KeNNKWPHo/gmj+JEyvimxRgXWYxxkcUYF/nwj0sZb/IoupTRpYzOo9hJGSc8iic8ip2UcbPFGBdZjHGRxRgX+XAZj6JLGZ1HsZMydjyKLmV0HkWXMnY8ii5lnPAoupTx/9lijIssxrjIYoyLfPiylPFNHsUTKaPzKLqUcSJl7HgUXcroPIouZXQexQmP4kTKOJEy/iaLMS6yGOMiizEu8uFlHsV/KWV0HkWXMjqPoksZnUfRpYzOo+hSRudRdCmj8yi6lNF5FF3K6DyKEymj8yhOeBR/s8UYF1mMcZHFGBf58FDK+Jd4FDsexYmU0XkUXcroPIouZXQexYmU8U0p41+yGOMiizEushjjIh++zKM4kTI6j6JLGd+UMnY8ihMeRZcyOo/iCY+iSxldythJGZ1HcSJl7HgUJ1LGE4sxLrIY4yKLMS7y4S+XMjqP4kTK2PEoOo/iTSmj8yh2UkbnUeykjM6j6FLGjkfRpYwdj+JEyvhJizEushjjIosxLvLhIY+iSxk7KWPHoziRMk54FF3KOOFRdCmj8yh2UsYTHkWXMnY8ip2UseNRdCmj8yh2UkbnUeykjCcWY1xkMcZFFmNc5MNDKeOER9GljJ2U8YRH0aWMzqPYSRk7HkWXMjqP4kTK6FLGjkfRpYzOo9jxKLqUcSJldB7FTsr4psUYF1mMcZHFGBf59fsPHvAo3pQyOo+iSxmdR7GTMk54FCdSxhMexRMpo/MoTqSMzqP4SSnjTYsxLrIY4yKLMS7y4aGUseNRdCmj8yh2UkbnUbzJo+hSxo5HseNRnEgZnUfRpYwnUsYTKeOER7GTMr5pMcZFFmNcZDHGRT485FF0KWPHo+hSRudR7KSMzqPoUsYTHkWXMp5IGZ1H0XkUJzyKEx7FmzyKLmXspIzOo9hJGU8sxrjIYoyLLMa4yIcvSxmdR7GTMt7kUeykjB2PoksZnUex41F0KaPzKLqU0XkUOyljx6M4kTJ2UkbnUXQpYydldB7FmxZjXGQxxkUWY1zkww9LGSc8ijeljBMpYydlnPAoupTReRRdyjjhUeykjB2PYidlnPAoftJijIssxrjIYoyL/Pr9By/yKE6kjBMexYmUseNR7KSMzqPoUsYJj+JEyug8ihMpY8ej6FLGEx7FTsr4psUYF1mMcZHFGBf58JfzKLqUseNRPJEyOo9ix6M4kTJ2PIrOo9hJGTsexQmPYidldB7FTsroPIqdlPHEYoyLLMa4yGKMi3x4WcrY8Sh2PIouZXQeRZcydjyKLmU8kTI6j6JLGZ1H0XkUXcrYSRmdR/FEyug8ii5lnEgZf5PFGBdZjHGRxRgX+fCQR/FNHsWJlNF5FJ1HcSJldB7FiZTReRQ7KWMnZex4FCdSRudRPJEyOo+iSxnftBjjIosxLrIY4yK/fv/BD/IodlLGjkdxImV8k0exkzJ2PIonUkbnUXQp400eRZcyOo/iRMp402KMiyzGuMhijIt8eMij2EkZT3gUJ1LGjkexkzJ2PIoTHsVOynjCo9jxKHZSxo5H0aWMnZTReRQ7HkWXMp5YjHGRxRgXWYxxkV+//+CLPIqdlNF5FF3K6DyKLmV0HkWXMp7wKHZSxgmPYidldB5FlzJ2PIouZZzwKLqU0XkUXco44VF0KeNNizEushjjIosxLvLhIY/imzyKLmXspIwdj2InZTzhUXQpYydl7KSMzqPYSRmdR9GljBMexY5HcSJldB5FlzKeWIxxkcUYF1mMcZFfv//gAY9iJ2W8yaPoUkbnUZxIGSc8ijeljM6j2EkZb/IoupSx41F0KWPHo+hSxjctxrjIYoyLLMa4yIeXpYzOoziRMjqPoksZOyljx6N4ImXseBRdyngiZXQexU/yKJ5IGZ1HsZMynliMcZHFGBdZjHGRD1+WMnY8is6j6FLGCY9iJ2Wc8Ch2UsaOR9GljM6j6FLGiZSx41F0KaPzKDqPoksZ/5LFGBdZjHGRxRgX+fDDPIoTHsWJlHHCoziRMr7Jo3jCo3iTR3EiZXQeRZcyOo/iTYsxLrIY4yKLMS7y4aGUcSJlvMmj2PEoTqSMzqN4ImXspIwTHsWJlNF5FDsp44RHsZMyOo+iSxlvWoxxkcUYF1mMcZEPD3kUPylldCmj8yi6lNF5FF3KeMKjOOFRnEgZXcrYSRk7KaPzKHY8ii5l7HgUXcrY8Si6lPHEYoyLLMa4yGKMi3x4Wcp4k0dxImXspIzOo+hSRpcyTqSMzqM4kTJOeBRdynhTyjiRMv5LizEushjjIosxLvLhyzyKEynjhEdxImV0KeOER9GljM6j6FLGCY9iJ2V0KaPzKJ7wKN7kUXQp45sWY1xkMcZFFmNc5MNlUsYTHkWXMnY8ii5l7HgUXcroUsYJj6JLGSc8ii5ldB5FlzI6j2InZfykxRgXWYxxkcUYF/kw/o+U0XkUJzyKJzyKLmXspIwdj+KER9GljM6j6FLGEx5FlzKeWIxxkcUYF1mMcZEPX5YyvilldB7FiZSxkzI6j+JEyug8is6j6FLGCY+iSxldyug8ip2U0XkU/5LFGBdZjHGRxRgX+fAyj+K/lDI6j+KER7GTMjqP4omUseNRdCljx6PYSRk7HkWXMjqPYsej+C8txrjIYoyLLMa4yK/ffzDGJRZjXGQxxkUWY1xkMcZFFmNcZDHGRRZjXGQxxkUWY1xkMcZFFmNcZDHGRRZjXGQxxkUWY1zkf3YCnz+XsBnPAAAAAElFTkSuQmCC)
https://github.com/hankei6km/mdast-qrcode
"
`;
